From e83bd5fe346267552c46d1dea66d1ecd612d300a Mon Sep 17 00:00:00 2001
From: Dan Kephart <dan.kephart@lairdtech.com>
Date: Tue, 28 Jun 2016 14:54:37 -0400
Subject: [PATCH] laird: ath6kl: sme_state shortcut to SME_DISCONNECTED removed

When an NL80211_DISCONNECT is sent to cfg80211, the driver's cfg80211
disconnect function sets the sme_state to SME_DISCONNECTED before receiving
a WMI_DISCONNECT_EVENT from the firmware.  This was causing the supplicant,
SDK events, and the SDK status function to not see that the connection is
currently disconnected becuase the driver fails to notify cfg80211 of the
successful disconnect.  This also caused the supplicant to fail to
reconnect after standby.

Bug: 6050 9085
---
 drivers/net/wireless/ath/ath6kl/cfg80211.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath6kl/cfg80211.c b/drivers/net/wireless/ath/ath6kl/cfg80211.c
index 88bd6a95f6c4..994f5ff0ed15 100644
--- a/drivers/net/wireless/ath/ath6kl/cfg80211.c
+++ b/drivers/net/wireless/ath/ath6kl/cfg80211.c
@@ -853,8 +853,6 @@ static int ath6kl_cfg80211_disconnect(struct wiphy *wiphy,
 
 	up(&ar->sem);
 
-	vif->sme_state = SME_DISCONNECTED;
-
 #ifdef CONFIG_ATH6KL_LAIRD_FIPS
 	if (fips_mode)
 		laird_setbssid(NULL);
-- 
2.11.0

